enhancements/update/to-multi-arch.md

```diff
@@ -0,0 +1,157 @@
+  - TBD
+creation-date: 2024-10-01
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/OTA-962
+see-also:
+  
+replaces:
+
+superseded-by:
+
+---
+
+# Upgrade To Multi Arch
+
+## Summary
+
+This enhancement defines the extension of openshift/api to improve the communication between CVO and cluster operators and how they use the API to evaluate the progress of [migrating an existing single-architecture cluster to multi-architecture](https://docs.openshift.com/container-platform/4.16/updating/updating_a_cluster/migrating-to-multi-payload.html#migrating-to-multi-payload).
+
+
+## Motivation
+
+During a cluster upgrade, the Cluster Version Operator (CVO) downloads the desired release payload and applies the manifests of OpenShift operators to the cluster.
+Each OpenShift operator reports the versions of its own and its operands via `ClusterOperator.status.versions` so that CVO can oversee the upgrade process of the operator.
+When all of their versions are the desired version, CVO considers that the upgrade of the operator completes.
+
+The command `oc adm upgrade --to-multi-arch` is used to upgrade the cluster from a single-architecture to a multi-architecture cluster *at the current version*.
+Because `ClusterOperator.status.versions` does not provide information about the multi-arch of the operator's image and `--to-multi-arch` starts the upgrade targeting *the same version*, the upgrade of each operator finishes right away from the CVO's point of view. Without the ability to recognize the component's turning into multi-arch, CVO's evaluation of the upgrade progress is inaccurate . This enhancement is to define a proposal to address the issue.
+
+
+### User Stories
+
+* As a cluster admin, I want to check the upgrade progress of migrating a cluster to multi-architecture.
+
+
+### Goals
+
+Provide a proposal that addresses the issue about the upgrade progress during a cluster migration to multi-arch.
+
+### Non-Goals
+
+
+## Proposal
+
+The proposal includes
+- the extension of `ClusterOperator.status.versions` and `ClusterVersion.status.desired` in openshift/api to provide information about the architecture, 
+- the implementation details on how an operator adopts the extension, and
+- the implementation details on how CVO adopts the extension.
+
+Any cluster operator, besides the existing fields `ClusterOperator.status.versions.version` and `ClusterOperator.status.versions.name`, may fill in the field `ClusterOperator.status.versions.architecture` with the value from the environment variable `OPERATOR_IMAGE_ARCHITECTURE` which are defined in the deployment of the operator like `OPERATOR_IMAGE_VERSION`. See [cluster-authentication-operator](https://github.com/openshift/cluster-authentication-operator/blob/0b7172707fcb3ba9880e8b100542700bd7d080d8/manifests/07_deployment.yaml#L69-L72) for example.
```

Comment:
I prefer links that pin a recent commit, as Hongkai has a the moment, because "I wonder if the dev branch looks like this" is selecting the default branch from the GitHub UI's drop-down in two clicks:

![image](https://github.com/user-attachments/assets/fcef74d4-b609-4aee-847b-0371e8f79f17)

While if there are floating links in the doc, and subsequent development in that branch shifts the content around, maybe `manifests/07_deployment.yaml` no longer exists and the link 404s.  Or maybe `#L69-L72` points to some content that is not the `env` entries Hongkai had been originally linking.  Trying to go back through the history of the repo to find a commit from when the doc-author might have constructed their floating URI is tedious, and hard to do reliably.

Reply:
OK, yes, you are right. Thanks!